### ISSUE TITLE
What is the geometric meaning of the process in "_do_paste_mask" function in "fcn_mask_head.py"

### ISSUE THREAD

**Issue by AkihiroSasabe** [open-mmlab/mmdetection#3745]
I'm sorry if I asked an out-of-place question.
I don't know what the geometric meaning of the following code is.

https://github.com/open-mmlab/mmdetection/blob/11b1ef830aa7c8c054d8b5346fe0966e643d473c/mmdet/models/roi_heads/mask_heads/fcn_mask_head.py#L295

```python
    img_y = (img_y - y0) / (y1 - y0) * 2 - 1
    img_x = (img_x - x0) / (x1 - x0) * 2 - 1
    # img_x, img_y have shapes (N, w), (N, h)
```

Why do you multiply `(img_y - y0) / (y1 - y0)` by `2`?
Do I need to multiply `1` if the mask size is 14x14 instead of 28x28, or `4` if it's 56x56?

I understand that `img_x` and `img_y` are needed to derive the grid, which is an argument to the `F.grid_sample` function.

And I also know that `img_masks` (100x1x427x640 for 'data/coco/val2017/000000397133.jpg') is obtained by overlaying 28x28 masks on the input image at the right place and the right size, using bilinear interpolation.

For example, for the inference in data/coco/val2017/000000397133.jpg: 
 - masks[0,0,:,:]
![image](https://user-images.githubusercontent.com/34682532/93014153-95585280-f5e9-11ea-8929-98e6ef43f297.png)

 - img_masks[0,0,:,:] 
![image](https://user-images.githubusercontent.com/34682532/93014177-dea8a200-f5e9-11ea-9d09-4c672146f499.png)

 - All bounding boxes and masks on input image(data/coco/val2017/000000397133.jpg): 
![image](https://user-images.githubusercontent.com/34682532/93014199-026be800-f5ea-11ea-97d0-6ab1f63f8b3f.png)

Thank you very much for your response.

**Comment by ZwwWayne** [comment]
`img_y = (img_y - y0) / (y1 - y0) ` means to rescale the coordinates of image into range [0, 1] and then rescale and shift it into 
 range [-1, 1]. See the doc of `F.grid_sample` [here](https://pytorch.org/docs/stable/nn.functional.html?highlight=grid_sample#torch.nn.functional.grid_sample), that is for the valid input range. The value in coordinates outside of the box will be set as zero. This maps the mask inside a box into the complete image mask.
